cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -1415,9 +1415,25 @@ queryResult eAcq = pure eAcq
 
 toEpochInfo :: EraHistory CardanoMode -> EpochInfo (Either Text)
 toEpochInfo (EraHistory _ interpreter) =
-  hoistEpochInfo (first (Text.pack . show ) . runExcept)
+  hoistEpochInfo (first (Text.pack . show) . runExcept)
     $ Consensus.interpreterToEpochInfo interpreter
 
+-- | A value that is tentative or produces a tentative value if used.  These values
+-- are considered accurate only if some future event such as a hard fork does not
+-- render them invalid.
+newtype Tentative a = Tentative { tentative :: a } deriving (Eq, Show)
+
+-- | Get an Epoch Info that computes tentative values.  The values computed are
+-- tentative because it uses an interpreter that is extended past the horizon.
+-- This interpreter will compute accurate values into the future as long as a
+-- a hard fork does not happen in the intervening time.  Those values are thus
+-- "tentative" because they can change in the event of a hard fork.
```

Comment:
They can change in the event of a hardfork occurring _after_ the horizon.  Normally the interpreter refuses to give an answer beyond the horizon so we construct a different interpreter that ignores the horizon.  

Values before the horizon won't change.

However, because the type allows for values beyond the horizon to change, it is annotated as `Tentative`.